tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set(CMAKE_ASM_COMPILER "arm-none-eabi-gcc")
+set(CMAKE_C_COMPILER "arm-none-eabi-gcc")
+set(CMAKE_CXX_COMPILER "arm-none-eabi-g++")
+set(GCC_ELF2BIN "arm-none-eabi-objcopy")
+set_property(GLOBAL PROPERTY ELF2BIN ${GCC_ELF2BIN})
+
+
+# Sets toolchain options
+function(mbed_set_toolchain_options target)
+    list(APPEND link_options
+        "-Wl,--start-group"
+            "-lstdc++"
+            "-lsupc++"
+            "-lm"
+            "-lc"
+            "-lgcc"
+            "-lnosys"
+        "-Wl,--end-group"
+        "-specs=nosys.specs"
+        "-T" "${CMAKE_BINARY_DIR}/${APP_TARGET}.link_script.ld"
+        "-Wl,-Map=${CMAKE_BINARY_DIR}/${APP_TARGET}.map"
+        "-Wl,--cref"
+    )
+
+    # Add linking time preprocessor macro for TFM targets
+    if("TFM" IN_LIST MBED_TARGET_LABELS)
+        list(APPEND link_options
+            "-DDOMAIN_NS=1"
+        )
+    endif()
+
+    list(APPEND common_options
+        "-Wall"
+        "-Wextra"
+        "-Wno-unused-parameter"
+        "-Wno-missing-field-initializers"
+        "-fmessage-length=0"
+        "-fno-exceptions"
+        "-ffunction-sections"
+        "-fdata-sections"
+        "-funsigned-char"
+        "-MMD"
+        "-fomit-frame-pointer"
+        "-g3"
+    )
+
+    target_compile_options(${target}
+        PUBLIC
+            ${common_options}
+    )
+
+    target_compile_definitions(${target}
+        PUBLIC
+            TOOLCHAIN_GCC_ARM
+            TOOLCHAIN_GCC
+    )
+
+
+    target_link_options(${target}
+        PUBLIC
+            ${common_options}
+            ${link_options}
+    )
+endfunction()
+
+# GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
+# this step - linker options and compile definitions 
+function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+    set(_compile_definitions 
+        "$<TARGET_PROPERTY:${target},COMPILE_DEFINITIONS>"
+    )
+
+    set(_linker_options
+        "$<TARGET_PROPERTY:${target},LINK_OPTIONS>"
+    )
```

Comment:
@0xc0170 
`LINK_OPTIONS` are mostly flags for linking when we trigger `arm-none-eabi-ld`). Are they relevant to _preprocessing_ which is basically just text manipulating?

Even if we do need link options, we haven't done any transformation. For compile definitions we join the list with ` -D`. But if we don't join the list, the output file `linker_options.txt` is semicolon-separated instead of space-separated.

Reply:
It is questionable, see

```
    # Add linking time preprocessor macro for TFM targets
    if("TFM" IN_LIST MBED_TARGET_LABELS)
        list(APPEND link_options
            "-DDOMAIN_NS=1"
        )
    endif()
```

true, we do not need most of the options but if some component adds a define like this, we would skip it. Or is there a better way?